index.js

```diff
@@ -337,88 +338,118 @@ class GitlabScm extends Scm {
-            parsed.lastCommitMessage = Hoek.reach(webhookPayload,
-                'commits.-1.message', { default: '' });
+            if (action === 'merged') {
+                action = 'closed';
+            }
 
-            return Promise.resolve(parsed);
+            return {
+                action,
+                branch: Hoek.reach(mergeRequest, 'target_branch'),
+                checkoutUrl,
+                prNum,
+                prTitle,
+                prRef: `merge_requests/${prNum}`,
+                ref,
+                prSource,
+                sha: Hoek.reach(mergeRequest, 'last_commit.id'),
+                type: 'pr',
+                username: Hoek.reach(webhookPayload, 'user.username'),
+                hookId,
+                scmContext
+            };
         }
-        case 'merge_request': {
-            const mergeRequest = webhookPayload.object_attributes;
-
-            if (mergeRequest.state === 'opened') {
-                parsed.action = 'opened';
-            } else if (mergeRequest.state === 'reopened') {
-                parsed.action = 'reopened';
-            } else if (mergeRequest.state === 'closed' || mergeRequest.state === 'merged') {
-                parsed.action = 'closed';
-            } else {
-                return Promise.resolve(null);
+        case 'push': {
+            if (Array.isArray(commits)) {
+                commits.forEach((commit) => {
+                    commitAuthors.push(commit.author.name);
+                });
             }
 
-            parsed.type = 'pr';
-            parsed.username = webhookPayload.user.username;
-            parsed.checkoutUrl = mergeRequest.source.git_http_url;
-            parsed.branch = mergeRequest.target_branch;
-            parsed.sha = mergeRequest.last_commit.id;
-            parsed.prNum = mergeRequest.iid;
-            parsed.prRef = `merge_requests/${mergeRequest.iid}`;
-
-            return Promise.resolve(parsed);
+            return {
+                action: 'push',
+                branch: Hoek.reach(webhookPayload, 'ref').split('/').slice(-1)[0],
+                checkoutUrl,
+                sha: Hoek.reach(webhookPayload, 'checkout_sha'),
+                type: 'repo',
+                username: Hoek.reach(webhookPayload, 'user_username'),
+                commitAuthors,
+                lastCommitMessage: Hoek.reach(webhookPayload, 'commits.-1.message',
+                    { default: '' }) || '',
+                hookId,
+                scmContext,
+                ref: Hoek.reach(webhookPayload, 'ref')
+            };
         }
         default:
-            return Promise.resolve(null);
+            logger.info('%s event is not available yet in scm-gitlab plugin', type);
+
+            return null;
         }
     }
 
     /**
      * Checkout the source code from a repository; resolves as an object with checkout commands
-     * @method getCheckoutCommand
+     * @async getCheckoutCommand
```

Comment:
is `async` tag needed https://jsdoc.app/tags-async.html
I don't see an `method` tag, though there is `function` https://jsdoc.app/

Reply:
not sure, my IDE always adds `@method` by default
saw that scm-github had `@async` so I was adding those to match